api/v1beta1/endpointservice_types.go

```diff
@@ -0,0 +1,90 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&AWSEndpointService{}, &AWSEndpointServiceList{})
+}
+
+// The following are reasons for the IgnitionEndpointAvailable condition.
+const (
+	// AWSEndpointServiceAvailable indicates whether the AWS Endpoint Service
+	// has been created for the specified NLB in the management VPC
+	AWSEndpointServiceAvailable ConditionType = "EndpointServiceAvailable"
+
+	// AWSEndpointServiceAvailable indicates whether the AWS Endpoint has been
+	// created in the guest VPC
+	AWSEndpointAvailable ConditionType = "EndpointAvailable"
+
+	AWSSuccessReason string = "AWSSuccess"
+	AWSErrorReason   string = "AWSError"
+)
+
+// AWSEndpointServiceSpec defines the desired state of AWSEndpointService
+type AWSEndpointServiceSpec struct {
+	// The name of the NLB for which an Endpoint Service should be configured
+	NetworkLoadBalancerName string `json:"networkLoadBalancerName"`
+
+	// SubnetIDs is the list of subnet IDs to which guest nodes can attach
+	// +optional
+	SubnetIDs []string `json:"subnetIDs,omitempty"`
+
+	// Tags to apply to the EndpointService
+	// +optional
+	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+}
+
+// AWSEndpointServiceStatus defines the observed state of AWSEndpointService
+type AWSEndpointServiceStatus struct {
+	// EndpointServiceName is the name of the Endpoint Service created in the
+	// management VPC
+	// +optional
+	EndpointServiceName string `json:"endpointServiceName,omitempty"`
+
+	// EndpointID is the ID of the Endpoint created in the guest VPC
+	// +optional
+	EndpointID string `json:"endpointID,omitempty"`
+
+	// Deprecated: Use DNSNames instead
```

Comment:
Aren't we dropping all deprecated fields?

Reply:
yes, need to remove